### PR TITLE
Fix a warning that shift overflows data type width

### PR DESCRIPTION
**Pull request by DennisOSRM**
This is a minor, non-functional change that fixes the following compiler warning on GCC8:

```
/home/dennis/Coding/rpcs3/Utilities/Thread.cpp: In function ‘bool handle_access_violation(u32, bool, x64_context*)’:
/home/dennis/Coding/rpcs3/Utilities/Thread.cpp:1271:66: warning: left shift count >= width of type [-Wshift-count-overflow]
      be_t<u64> data2 = (SYS_MEMORY_PAGE_FAULT_TYPE_PPU_THREAD << 32) + cpu->id; // TODO: fix hack for now that assumes PPU thread always.
```

### CODE REVIEW

File: Utilities/Thread.cpp [hunk 1]
```diff
@@ -1268,7 +1268,7 @@ bool handle_access_violation(u32 addr, bool is_writing, x64_context* context)
 					// Now, we notify the game that a page fault occurred so it can rectify it.
 					// Note, for data3, were the memory readable AND we got a page fault, it must be due to a write violation since reads are allowed.
 					be_t<u64> data1 = addr;
-					be_t<u64> data2 = (SYS_MEMORY_PAGE_FAULT_TYPE_PPU_THREAD << 32) + cpu->id; // TODO: fix hack for now that assumes PPU thread always.
+					be_t<u64> data2 = ((be_t<u64>)SYS_MEMORY_PAGE_FAULT_TYPE_PPU_THREAD << 32) + cpu->id; // TODO: fix hack for now that assumes PPU thread always.
```
Comment (elad335) [inline Utilities/Thread.cpp]: while at it, fix the type of `data1` to `data3`, it should be u64. (same for SYS_MEMORY_PAGE_FAULT_TYPE_PPU_THREAD)

Comment (DennisOSRM) [reply Utilities/Thread.cpp]: done

File: rpcs3/Emu/Cell/lv2/sys_mmapper.h [hunk 2]
```diff
@@ -23,13 +23,13 @@ struct lv2_memory : lv2_obj
...
 {
 	SYS_MEMORY_PAGE_FAULT_EVENT_KEY	       = 0xfffe000000000000ULL,
+	SYS_MEMORY_PAGE_FAULT_TYPE_PPU_THREAD  = 0x0ULL,
 };
 
 enum : u32
 {
 	SYS_MEMORY_PAGE_FAULT_CAUSE_NON_MAPPED = 0x00000002U,
 	SYS_MEMORY_PAGE_FAULT_CAUSE_READ_ONLY  = 0x00000001U,
-	SYS_MEMORY_PAGE_FAULT_TYPE_PPU_THREAD  = 0x00000000U,
```
Comment (elad335) [inline rpcs3/Emu/Cell/lv2/sys_mmapper.h]: This constant should still be here.

Comment (elad335) [reply rpcs3/Emu/Cell/lv2/sys_mmapper.h]: better to just make the whole enum u64

Comment (DennisOSRM) [reply rpcs3/Emu/Cell/lv2/sys_mmapper.h]: gotcha

**Pull request merged**